sdcm/utils/gce_utils.py

```diff
@@ -271,3 +262,245 @@ def _get_log_entries(self, service, query, page_token=None):
+            used subnet will be used.
+        external_access: boolean flag indicating if the instance should have an external IPv4
+            address assigned.
+        external_ipv4: external IPv4 address to be assigned to this instance. If you specify
+            an external IP address, it must live in the same region as the zone of the instance.
+            This setting requires `external_access` to be set to True to work.
+        accelerators: a list of AcceleratorConfig objects describing the accelerators that will
+            be attached to the new instance.
+        preemptible: boolean value indicating if the new instance should be preemptible
+            or not. Preemptible VMs have been deprecated and you should now use Spot VMs.
+        spot: boolean value indicating if the new instance should be a Spot VM or not.
+        instance_termination_action: What action should be taken once a Spot VM is terminated.
+            Possible values: "STOP", "DELETE"
+        custom_hostname: Custom hostname of the new VM instance.
+            Custom hostnames must conform to RFC 1035 requirements for valid hostnames.
+        delete_protection: boolean value indicating if the new virtual machine should be
+            protected against deletion or not.
+        network_tags: List of tags to apply to network labels
+        metadata: dict of key values to add to metadata
+        service_accounts: list of service account to attach to the instance
+    Returns:
+        Instance object.
+    """
+    instance_client, _ = get_gce_compute_instances_client()
+
+    # Use the network interface provided in the network_link argument.
+    network_interface = compute_v1.NetworkInterface()
+    network_interface.network = f"global/networks/{network_name}"
+    if subnetwork_link:
+        network_interface.subnetwork = subnetwork_link
+
+    if internal_ip:
+        network_interface.network_i_p = internal_ip
+
+    if external_access:
+        access = compute_v1.AccessConfig()
+        access.type_ = compute_v1.AccessConfig.Type.ONE_TO_ONE_NAT.name
+        access.name = "External NAT"
+        access.network_tier = access.NetworkTier.PREMIUM.name
+        if external_ipv4:
+            access.nat_i_p = external_ipv4
+        network_interface.access_configs = [access]
+
+    # Collect information into the Instance object.
+    instance = compute_v1.Instance()
+    instance.network_interfaces = [network_interface]
+    instance.name = instance_name
+    instance.disks = disks
+    if re.match(r"^zones/[a-z\d\-]+/machineTypes/[a-z\d\-]+$", machine_type):
+        instance.machine_type = machine_type
+    else:
+        instance.machine_type = f"zones/{zone}/machineTypes/{machine_type}"
+
+    if accelerators:
+        instance.guest_accelerators = accelerators
+
+    if preemptible:
+        # Set the preemptible setting
+        warnings.warn(
+            "Preemptible VMs are being replaced by Spot VMs.", DeprecationWarning
```

Comment:
As I understand the logic, the message must be different.
It must be something like following:
```'preemptible' option is deprecated, use 'spot' instead.```

The current message, not seeing code, makes me think that some is in process of replacement right now.
Which is false.

Then, since we add new code, why adding the deprecated option in general?

Reply:
it's actually copy paste form google own code sample, it mean the feature "Preemptible VMs" is replace with "Spot VMs"

anyhow we need this code since we now use preemptible, and we want to switch to using spots, we can ditch this warning.
it doesn't really helps us.

Reply:
I've removed the misleading warning